packages/ssi-sdk-core/src/utils/image.ts

```diff
@@ -1,25 +1,67 @@
 import fetch from 'cross-fetch'
-import sizeOf from 'image-size'
-import { ISizeCalculationResult } from 'image-size/dist/types/interface'
+import { imageSize } from 'image-size'
 import { IImageDimensions, IImageResource } from '../types'
 
+type SizeCalculationResult = {
+  width?: number
+  height?: number
+  orientation?: number
+  type?: string
+}
+
+const uint8ArrayToString = (uint8Array: Uint8Array): string => {
+  const decoder = new TextDecoder('utf-8')
+  return decoder.decode(uint8Array)
+}
+
+// TODO: here we're handling svg separately, remove this section when image-size starts supporting it in version 2
+const isSvg = (uint8Array: Uint8Array): boolean => {
+  const text = uint8ArrayToString(uint8Array)
+  const normalizedText = text.trim().toLowerCase()
+  return normalizedText.startsWith('<svg') || normalizedText.startsWith('<?xml')
+}
+
+const getSvgDimensions = (uint8Array: Uint8Array) => {
+  const svgContent = uint8ArrayToString(uint8Array)
+  const widthMatch = svgContent.match(/width="([^"]+)"/)
```

Comment:
Does this work at all? For a simple SVG probably it might, but what about an SVG with all kinds of objects in it?

Reply:
changed this method and wrote 2 additional tests for a multi-layered and a animated svg